File: slideholder/src/main/java/zhan/library/slide/ISlideHelper.java
package zhan.library.slide;

import java.util.ArrayList;
import java.util.List;

public class ISlideHelper {

    private List<ISlide> mISlides = new ArrayList<>();

    public void add(ISlide iSlide) {
        mISlides.add(iSlide);
    }

    public void remove(ISlide iSlide) {
        mISlides.remove(iSlide);
    }

    public void clear() {
        mISlides.clear();
    }

    public List<ISlide> getISlideList() {
        return mISlides;
    }

    public void slideOpen() {
        for (ISlide slide : mISlides) {
            slide.slideOpen();
        }
    }

    public void slideClose() {
        for (ISlide slide : mISlides) {
            slide.slideClose();
        }
    }
}

File: slideholder/src/main/java/zhan/library/slide/ISlideHelper.kt
package zhan.library.slide

import java.util.ArrayList

class ISlideHelper {

    private val mISlides = ArrayList<ISlide>()

    val iSlideList: List<ISlide>
        get() = mISlides

    fun add(iSlide: ISlide) {
        mISlides.add(iSlide)
    }

    fun remove(iSlide: ISlide) {
        mISlides.remove(iSlide)
    }

    fun clear() {
        mISlides.clear()
    }

    fun slideOpen() {
        for (slide in mISlides) {
            slide.slideOpen()
        }
    }

    fun slideClose() {
        for (slide in mISlides) {
            slide.slideClose()
        }
    }
}
